Security section for Umbraco-Cloud (#2319)
umbraco/UmbracoDocs: Umbraco-Cloud/Security/index.md

# Security on Umbraco Cloud

## HTTPS & Certificates

> All Umbraco Cloud websites use HTTPS by default. Both the default {projectName}.s1.umbraco.io and custom domains are protected by periodically renewed certificates from Let's Encrypt. This service is offered as part of Umbraco Cloud by [Umbraco LATCH](/Umbraco-Cloud/Set-Up/Umbraco-Latch/index.md)

Reviewer: ```suggestion
All Umbraco Cloud websites use HTTPS by default. Both the default {projectName}.s1.umbraco.io and custom domains are protected by periodically renewed certificates from Let's Encrypt. This service is offered as part of Umbraco Cloud by [Umbraco LATCH](/Umbraco-Cloud/Set-Up/Umbraco-Latch).
```

### Custom Certificates

> Custom certificates can be used with all custom domains. Please refer to [Managing Custom Certificates documentation](Umbraco-Cloud/Set-Up/Manage-Hostnames/Security-Certificates/index.md)

Reviewer: ```suggestion
Custom certificates can be used with all custom domains. Please refer to our [Managing Custom Certificates documentation](Umbraco-Cloud/Set-Up/Manage-Hostnames/Security-Certificates)
```

### TLS support

> As of April 2020, we've deprecated support for TLS1.0 & TLS 1.1. TLS1.2 is the only supported TLS protocol going forward.

Reviewer 2: Should this say "minimum supported version" ?

Reviewer 2: Should we explicitly state that we add support for TLS 1.2 for the older .net framework versions where TLS 1.0 was the default and TLS 1.2 wasn't added by default? I believe this is the case for sites on Cloud running Umbraco 7.x and .net framework 4.5.2.

Author: @sitereactor we only support TLS1.2, we don't support 1.3, that means 1.2 is the only supported version. We could remove "only".

I've now specifically mentioned changing the default TLS version for version 7 websites.

Reviewer 2: Sure I get that. I just thought that "minimum" would be better, so we don't have something that is potentially wrong for certain parts of the platform when the hosting changes or when we have a better DNS/CDN provider in front. But fine to leave this for now.

> As of April 2020, we've deprecated support for TLS1.0 & TLS 1.1. TLS1.2 is the only supported TLS protocol going forward.

Reviewer: ```suggestion
As of April 2020, we've deprecated support for TLS 1.0 & TLS 1.1. TLS 1.2 is the only supported TLS protocol going forward.
```

### TLS Ciphers support

Umbraco Cloud Websites support the following TLS ciphers in this order:

- TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256
- TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA
- TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA
- TLS_DHE_RSA_WITH_AES_256_GCM_SHA384
- TLS_DHE_RSA_WITH_AES_128_GCM_SHA256
- TLS_DHE_RSA_WITH_AES_256_CBC_SHA
- TLS_DHE_RSA_WITH_AES_128_CBC_SHA

> ### TLS1.2 by default in external services

Reviewer: ```suggestion
### TLS 1.2 by default in external services
```

> In order to integrate older external applications accessing Umbraco Cloud Websites you might have to modify the TLS support in the .Net application.

Reviewer: ```suggestion
In order to integrate older external applications by accessing Umbraco Cloud Websites you might have to modify the TLS support in the .NET application.
```

Author: It's the external applications which are accessing Umbraco Cloud Websites. Adding "by" changes the meaning. Should we change this into something else then?

Reviewer: Ah, I see 🤔
Perheps we could say `to` instead?
```suggestion
In order to integrate older external applications to access Umbraco Cloud Websites you might have to modify the TLS support in the .Net application.
```

That keeps the origin meaning, right?

> For ASP.NET applications, inspect the `<system.web><httpRuntime targetFramework>` element of web.config to find the version of the .NET Framework your application is using. .NET applications on .NET 4.7+ are using the OS specified TLS protocols. In Windows 8 & 10, Windows Server 2012 & 2016 that's 1.2+ by default, therefore no actions necessary. .NET applications lower then 4.7 require updates to ensure they can communicate using TLS1.2 by default.

Reviewer: ```suggestion
For ASP.NET applications, inspect the `<system.web><httpRuntime targetFramework>` element of web.config to find the version of the .NET Framework your application is using. .NET applications on .NET 4.7+ are using the OS specified TLS protocols. In Windows 8 & 10, Windows Server 2012 & 2016 TLS 1.2+ is used by default, therefore no actions necessary. .NET applications lower then 4.7 require updates to ensure they can communicate using TLS 1.2 by default.
```

> More information specifically from MS about .Net applications and TLS support can be found [in MS docs](https://docs.microsoft.com/en-us/dotnet/framework/network-programming/tls#audit-your-code-and-make-code-changes). For other application frameworks/languages we encourage to lookup their respective documentations.

Reviewer 2: write Microsoft rather then MS

Author: Did so.

### HTTP

> HTTP protocol is supported but not used by default on Umbraco Cloud Websites. If you'd like to keep using HTTP, which we strongly discourage, you'll need to remove a web.config transform as specified in [Umbraco LATCH documentation](/Umbraco-Cloud/Set-Up/Umbraco-Latch/index.md)

Reviewer: ```suggestion
HTTP protocol is supported but not used by default on Umbraco Cloud Websites. If you'd like to keep using HTTP, which we strongly discourage, you'll need to remove a web.config transform as specified in [Umbraco LATCH documentation](/Umbraco-Cloud/Set-Up/Umbraco-Latch)
```
